transit_odp/fares_validator/views/functions.py

```diff
@@ -111,6 +111,116 @@ def check_operator_id_format(context, operators, *args):
     return False
 
 
+def check_type_of_frame_ref_ref(context, type_of_frame_ref, *args):
+    """
+    Check if TypeOfFrameRef has either UK_PI_FARE_PRODUCT or
+    UK_PI_FARE_PRICE in it.
+    """
+    try:
+        type_of_frame_ref_ref = _extract_attribute(type_of_frame_ref, "ref")
+    except KeyError:
+        return False
+    for ref_value in FAREFRAME_TYPE_OF_FRAME_REF_SUBSTRING:
+        if ref_value in type_of_frame_ref_ref:
```

Comment:
This will always fail for some element where the value of ref is not in that list